### PR TITLE
[REVIEW] Remove UNKNOWN_NULL_COUNT workaround from contiguous_split()

### PR DESCRIPTION
**Pull request by nvdbaranec**
contiguous_split() had a small hack in it to get around an issue with slice().  Slice was returning UNKNOWN_NULL_COUNT for all sliced columns.  This caused computation of that number silently within contiguous_split() which caused performance issues because of the scale contiguous_split runs at (ten of thousands or more columns).  

slice() has been changed to produce proper null counts (https://github.com/rapidsai/cudf/issues/3600) so the hack is no longer necessary.

### CODE REVIEW

**Comment by nvdbaranec** [comment]
rerun tests

File: cpp/src/copying/contiguous_split.cu [hunk 1]
```diff
@@ -243,36 +243,22 @@ struct column_copy_functor {
...
-      //            
-      // Remove this hack once rapidsai/cudf#3600 is fixed.
-      column_view   in_wrapped{in.type(), in.size(), in.head<T>(), 
-                               in.null_mask(), in.null_mask() == nullptr ? UNKNOWN_NULL_COUNT : 0,
-                               in.offset() };
-      mutable_column_view  mcv{in.type(), in.size(), data, 
-                               validity, validity == nullptr ? UNKNOWN_NULL_COUNT : 0 };
+      // output copied column
+      mutable_column_view  mcv{in.type(), in.size(), data, validity, in.null_count()};
       if(in.nullable()){
```
Comment (karthikeyann) [inline cpp/src/copying/contiguous_split.cu]: consider  `in.has_nulls()` instead of `in.nullable()`

**Comment by karthikeyann** [comment]
rerun tests

File: cpp/src/copying/contiguous_split.cu (hunk 1 above)
Comment (nvdbaranec) [reply cpp/src/copying/contiguous_split.cu]: Yep, that makes sense.  Especially in this case, where we're now guaranteed to have a valid null count.  I'll change.

**Comment by jrhemstad** [comment]
rerun tests

**Comment by nvdbaranec** [comment]
Changed code to use has_nulls() instead of nullable() where appropriate.  This necessitated adding cudf::test::expect_tables_equivalent() which wraps expect_columns_equal().

File: cpp/tests/utilities/table_utilities.cu [hunk 2]
```diff
@@ -19,5 +19,15 @@ void expect_tables_equal(cudf::table_view lhs, cudf::table_view rhs) {
   }
 }
 
+/**
+ * @copydoc cudf::test::expect_tables_equivalent
+ * 
+ **/
+void expect_tables_equivalent(cudf::table_view lhs, cudf::table_view rhs) {  
+  for (auto i=0; i<lhs.num_columns(); ++i) {
```
Comment (aschaffer) [inline cpp/tests/utilities/table_utilities.cu]: `lhs.num_columns()` may be called repeatedly in this loop (unless there is a guarantee the compiler can cache the `num_calls()` call, in which case please ignore this comment). You might consider one of the following:

1. store the `num_columns()` result before using it;
2. use (const) iterators; 
3. if possible, use range-based loops (probably not possible, without additional expensive machinery);
4. use `std::for_each()`

Comment (nvdbaranec) [reply cpp/tests/utilities/table_utilities.cu]: Updated.  Note :  there's no for_each() that takes two iterator ranges, unfortunately.

Comment (aschaffer) [reply cpp/tests/utilities/table_utilities.cu]: `std::transform()` does (...take 2 ranges)

**Pull request merged**